app/controllers/tips_controller.rb

```diff
@@ -1,12 +1,8 @@
 class TipsController < ApplicationController
 	def index
-		@trip = Trip.find(params[:trip_id])
-		@tips = @trip.tips
-
-		respond_to do |format|
-			format.html { render }
-			format.json { render json: @tips }
-		end
+    @user = User.find_by(id: params[:user_id])
+    @trip = @user.trips.find_by(id: params[:trip_id])
```

Comment:
Part of it was habit. Since we are getting passed both the user and the trip ids we should use them. Its  a double check to make sure that the user in the link actually owns the trip in the link. This would work for friends as well. When we create the friends trip links for the page the user id in that link would be the friend not the current user.